### PR TITLE
Restore status for identities. Fix tests.

### PR DESCRIPTION
**Pull request by piohei**
## Motivation

We need to revert status field as it is used by apps. Some changes also brings back original app behaviour and responses.

## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation

### CODE REVIEW

File: e2e_tests/scenarios/tests/common/api.rs [hunk 1]
```diff
@@ -116,21 +116,32 @@ pub async fn inclusion_proof_raw(
...
         .expect("Could not parse response bytes to utf-8");
 
-    Ok(RawResponse {
+    let raw_response = RawResponse {
         status_code: response.status(),
         body: result,
-    })
+    };
+
+    debug!(
```
Comment (Dzejkop) [inline e2e_tests/scenarios/tests/common/api.rs:124]: do we need it?

File: src/server/data.rs [hunk 2]
```diff
@@ -1,19 +1,30 @@
...
 use semaphore::Field;
 use serde::{Deserialize, Serialize};
 
-use crate::identity_tree::{Hash, InclusionProof, RootItem};
+use crate::identity_tree::{Hash, InclusionProof, ProcessedStatus, RootItem, Status};
 use crate::prover::{ProverConfig, ProverType};
 
 #[derive(Debug, Serialize, Deserialize, PartialEq)]
-pub struct InclusionProofResponse(pub InclusionProof);
+pub struct InclusionProofResponse {
```
Comment (Dzejkop) [inline src/server/data.rs:11]: Isn't this type identitcal to `InclusionProof`?

File: e2e_tests/scenarios/tests/common/api.rs (hunk 1 above)
Comment (piohei) [reply e2e_tests/scenarios/tests/common/api.rs:124]: It was helpful for me in general when looking what happened on errors. That's why it is on debug level - to not show it always. Do you think I should remove only that one or other debug logs there also?

File: src/server/data.rs (hunk 2 above)
Comment (piohei) [reply src/server/data.rs:11]: It is but I think it is beneficial to keep this two domains (API and internal tree) not mixed. It is much easier to not brake the API by accident as we have customers that rely on it. That's why I also changed this way VerifySemaphoreProofResponse. Now hiding status is also not called as a function that someone may miss or forget to use. It is always done when converting from VerifySemaphoreProof. For me it is also much cleaner now what is happening there and when. :) I've found it hard when reverting that status and trying to ensure we are consistent with previous behaviour used by other teams.

Comment (Dzejkop) [reply src/server/data.rs:11]: Ok, sounds good

**Review by Dzejkop — APPROVED** [review]

**Pull request merged**